scripts/face_editor_extension.py

```diff
@@ -30,11 +31,26 @@ def before_process_batch(self, p, enabled: bool, *args, **kwargs):
         if not option.save_original_image:
             p.do_not_save_samples = True
 
-        if p.scripts is not None and hasattr(p.scripts, "alwayson_scripts") and p.scripts.alwayson_scripts[-1] != self:
-            for i, e in enumerate(p.scripts.alwayson_scripts):
-                if e == self:
-                    p.scripts.alwayson_scripts.append(p.scripts.alwayson_scripts.pop(i))
-                    break
+        if p.scripts is None and not hasattr(p.scripts, "alwayson_scripts"):
+            return
+        
+        script_index = shared.opts.data.get("face_editor_script_index", -1)
+        if script_index >= len(p.scripts.alwayson_scripts) or script_index < -len(p.scripts.alwayson_scripts):
+            return
+        
+        if p.scripts.alwayson_scripts[script_index] == self:
+            return
+        
+        for i, e in enumerate(p.scripts.alwayson_scripts):
+            if e != self:
+                continue
+
+            p.scripts.alwayson_scripts.pop(i)
+            if script_index == -1:
+                p.scripts.alwayson_scripts.append(self)
+            else:
+                p.scripts.alwayson_scripts.insert(script_index + 1, self)
+
```

Comment:
The code is a bit difficult to follow...

How about "the default value is `99` instead of `-1`"? Then I thought that the objective could be achieved with minimal modifications as follows.

```python
        if p.scripts is not None and hasattr(p.scripts, "alwayson_scripts"):
            script_index = shared.opts.data.get("face_editor_script_index", 99)
            for i, e in enumerate(p.scripts.alwayson_scripts):
                if e == self:
                    p.scripts.alwayson_scripts.insert(script_index, p.scripts.alwayson_scripts.pop(i))
                    break
```

Also, considering the intention of "execute as later as possible," I thought that the number `99` would be easier to understand than `-1`.

How about it?

Reply:
I was also struggling when writing this part, always thinking with a programmer's mindset. Your implementation is very appropriate, from the user's perspective, it is easy to understand and implement.